src/PointwiseFunctions/AnalyticData/CurvedWaveEquation/PureSphericalHarmonic.hpp

```diff
@@ -0,0 +1,110 @@
+ *
+ * \details The initial data is taken from \cite Scheel2003vs and sets the
+ * evolved variables of the scalar wave as follows:
+ *
+ * \f{align}
+ * \Psi &= 0 \\
+ * \Phi_i &= 0 \\
+ * \Pi &= \Pi_0(r, \theta, \phi) =  e^{- (r - r_0)^2 / w^2} Y_{lm}(\theta,
+ * \phi), \f}
+ *
+ * where \f$r_0\f$ is the radius of the profile and \f$w\f$ is its width. This
+ * describes a pure spherical harmonic mode \f$Y_{lm}(\theta, \phi)\f$ truncated
+ * by a circular Gaussian window function.
+ *
+ * When evolved, the scalar field \f$\Phi\f$ will briefly build up around the
+ * radius \f$r_0\f$ and then disperse. This can be used to study the ringdown
+ * behavior and late-time tails in different background spacetimes.
+ */
+
+class PureSphericalHarmonic : public MarkAsAnalyticData {
+ public:
+  struct Radius {
+    using type = double;
+    static constexpr Options::String help = {
+        "The radius of the spherical harmonic profile"};
+    static type lower_bound() noexcept { return 0.0; }
+  };
+
+  struct Width {
+    using type = double;
+    static constexpr Options::String help = {
+        "The width of the spherical harmonic profile."};
+    static type lower_bound() noexcept { return 0.0; }
+  };
+
+  struct l_mode {
+    using type = size_t;
+    static constexpr Options::String help = {
+        "The l-mode of the spherical harmonic Ylm"};
+  };
+
+  struct m_mode {
+    using type = double;
+    static constexpr Options::String help = {
+        "The m-mode of the spherical harmonic Ylm."};
+  };
+
+  using options = tmpl::list<Radius, Width, l_mode, m_mode>;
+
+  static constexpr Options::String help = {
+      "Initial data for a pure spherical harmonic mode truncated by a circular "
+      "Gaussian window funtion. The expression is taken from Scheel(2003), "
+      "equations 4.1-4.3."};
+
+  PureSphericalHarmonic() = default;
+  PureSphericalHarmonic(const PureSphericalHarmonic& /*rhs*/) = delete;
+  PureSphericalHarmonic& operator=(const PureSphericalHarmonic& /*rhs*/) =
+      delete;
+  PureSphericalHarmonic(PureSphericalHarmonic&& /*rhs*/) noexcept = default;
+  PureSphericalHarmonic& operator=(PureSphericalHarmonic&& /*rhs*/) noexcept =
+      default;
+  ~PureSphericalHarmonic() = default;
+
+  PureSphericalHarmonic(double radius, double width, size_t l, int m,
+                        const Options::Context& context = {});
+
+  explicit PureSphericalHarmonic(CkMigrateMessage* /*unused*/) noexcept {}
+
+  /// Retrieve the evolution variables at spatial coordinates `x`
+  tuples::TaggedTuple<CurvedScalarWave::Pi, CurvedScalarWave::Phi<3>,
+                      CurvedScalarWave::Psi>
+  variables(const tnsr::I<DataVector, 3>& x, double /*t*/,
+            tmpl::list<CurvedScalarWave::Pi, CurvedScalarWave::Phi<3>,
+                       CurvedScalarWave::Psi> /*meta*/) const noexcept;
+
+  // clang-tidy: no runtime references
+  void pup(PUP::er& /*p*/) noexcept;  //  NOLINT
+
+ private:
+  double radius_{};
```

Comment:
Always default-construct classes to an invalid state (NaN in this case). I've had hard-to-find bugs with classes that default-construct to a valid state even when they were forgotten in a pup function, so they did something different than expected.